velox/docs/develop/window.rst

```diff
@@ -0,0 +1,198 @@
+========================
+Window Operator
+========================
+
+Velox supports window function evaluation using Window operator. In this guide
+we will discuss some intricate design issues in this operator.
+
+
+Window frames
+-------------
+Window functions can optionally include a FRAME clause. The FRAME clause
+can be thought of as a spec for a sliding window of rows which bound
+the window function computation for the given row.
+
+Not all window functions are bounded by the FRAME clause. Aggregates
+computed as window functions and value functions :func:`first_value`,
+:func:`last_value` and :func:`nth_value` require window frames for
+evaluation.
+
+A frame can be ROWS type or RANGE type, and it runs from frame_start to
+frame_end. So a FRAME clause is one of
+
+.. code-block::
+
+   {RANGE|ROWS} frame_start
+   {RANGE|ROWS} BETWEEN frame_start AND frame_end
+
+frame_start and frame_end can be any of:
+
+.. code-block::
+
+   UNBOUNDED PRECEDING
+   expression PRECEDING
+   CURRENT ROW
+   expression FOLLOWING
+   UNBOUNDED FOLLOWING
+
+**ROWS mode**
+
+ROWS mode can be interpret as indices of the rows in the order in which they
+appear in the window partition. This ordering is determined by the ORDER BY
+clause. In ROWS mode, CURRENT ROW refers to the present row at which the
+function is being evaluated. Each consecutive row has increasing frame number.
+The frame numbers increase by 1 for each row.
+
+**RANGE mode**
+
+In RANGE mode, all peer rows have the same frame number.
+Rows are peers if they have the same values for the ORDER BY fields.
+A frame start of CURRENT ROW refers to the first peer row of the current row,
+while a frame end of CURRENT ROW refers to the last peer row of the current row.
+If no ORDER BY is specified, all rows are considered peers of the current row.
+
```

Comment:
Is it necessary to mention how to handle NULL if it exists in ORDER BY column?

Reply:
@PHILO-HE : Only Range frames are affected by NULL values. I am planning to add another section to this doc for k range frames. I'll give more details about NULL values in that section.